reflex/experimental/vars/base.py

```diff
@@ -59,12 +70,25 @@ def _var_full_name_needs_state_prefix(self) -> bool:
     def __post_init__(self):
         """Post-initialize the var."""
         # Decode any inline Var markup and apply it to the instance
-        _var_data, _var_name = _decode_var(self._var_name)
+        _var_data, _var_name = _decode_var_immutable(self._var_name)
         if _var_data:
             self.__init__(
-                _var_name, self._var_type, VarData.merge(self._var_data, _var_data)
+                _var_name,
+                self._var_type,
+                ImmutableVarData.merge(self._var_data, _var_data),
             )
 
+    def __hash__(self) -> int:
+        """Define a hash function for the var.
+
+        Returns:
+            The hash of the var.
+        """
+        return hash((self._var_name, self._var_type, self._var_data))
```

Comment:
do we really need this defined here? my concern is that some subclass adds additional fields, but then doesn't override `__hash__` and thus would compare the same as another Var, even though it really was different.

maybe that's not an issue though. i know `__hash__` is a bit weird, in that it is used for `set`/`dict` operations, but then they use equality for collision detection

Reply:
If it's not defined here we would have to define it elsewhere anyways, so I don't particularly see a way out of it.

Also `__eq__` is generally made to use the weird system of comparison to serialize it for the front end instead of actually comparing the two and returning a bool. It feels like this could cause a limitation between true equality and front-end equality.